### ISSUE TITLE
False positive in clone() type inference

### ISSUE THREAD

**Issue by la10736** [intellij-rust/intellij-rust#3593]
## Environment

* **Intellij-Rust plugin version:** v0.2.92.2116-183
* **Rust toolchain version:** both 1.29.0 and 1.32.0
* **IDE name and version:** 2018.3.5
* **Operating system:** Ubuntu 16.04

## Problem description

Wrong error message 

```rust
        impl<'ast> Visit<'ast> for TestFunctions {
            fn visit_item_fn(&mut self, item_fn: &'ast ItemFn) {
                if Self::is_test_fn(item_fn) {
                    self.0.push(item_fn.clone())
                }
            }
        }
```
show a 
>> mismatched types [E0308] expected `ItemFn`, found `&ItemFn` more... (Ctrl+F1)

on  `item_fn.clone()`

## Steps to reproduce

I wasn't able to reproduce it in a simpler project (without use `syn::visit::Visit`). Anyway you can find the code in https://github.com/la10736/rstest repo: unitest in `lib.rs` module.

**Comment by Kobzol** [comment]
The false positive is no longer present in the latest version of the plugin.
